ihmc-high-level-behaviors/src/libgdx/java/us/ihmc/rdx/perception/RDXStancePoseSelectionPanel.java

```diff
@@ -6,53 +6,72 @@
 import com.badlogic.gdx.graphics.g3d.RenderableProvider;
 import com.badlogic.gdx.utils.Array;
 import com.badlogic.gdx.utils.Pool;
+import ihmc_common_msgs.msg.dds.PoseListMessage;
 import imgui.ImGui;
 import imgui.flag.ImGuiMouseButton;
+import imgui.type.ImBoolean;
 import us.ihmc.behaviors.activeMapping.StancePoseCalculator;
+import us.ihmc.communication.packets.MessageTools;
+import us.ihmc.communication.ros2.ROS2Helper;
+import us.ihmc.euclid.geometry.Pose3D;
 import us.ihmc.euclid.referenceFrame.FramePose3D;
 import us.ihmc.euclid.referenceFrame.ReferenceFrame;
 import us.ihmc.euclid.tuple2D.Point2D;
 import us.ihmc.euclid.tuple3D.interfaces.Point3DReadOnly;
+import us.ihmc.footstepPlanning.communication.ContinuousWalkingAPI;
 import us.ihmc.footstepPlanning.graphSearch.FootstepPlannerEnvironmentHandler;
 import us.ihmc.footstepPlanning.tools.PlannerTools;
 import us.ihmc.perception.heightMap.TerrainMapData;
 import us.ihmc.perception.tools.PerceptionDebugTools;
 import us.ihmc.rdx.imgui.ImGuiTools;
+import us.ihmc.rdx.imgui.ImGuiUniqueLabelMap;
 import us.ihmc.rdx.imgui.RDXPanel;
 import us.ihmc.rdx.input.ImGui3DViewInput;
 import us.ihmc.rdx.tools.LibGDXTools;
 import us.ihmc.rdx.tools.RDXModelBuilder;
+import us.ihmc.rdx.ui.RDXStoredPropertySetTuner;
 import us.ihmc.rdx.ui.graphics.RDXFootstepGraphic;
 import us.ihmc.robotics.robotSide.RobotSide;
 import us.ihmc.robotics.robotSide.SegmentDependentList;
 import us.ihmc.robotics.robotSide.SideDependentList;
 import us.ihmc.sensorProcessing.heightMap.HeightMapData;
 
 import java.util.ArrayList;
+import java.util.List;
 
 public class RDXStancePoseSelectionPanel extends RDXPanel implements RenderableProvider
 {
+   private final ImGuiUniqueLabelMap labels = new ImGuiUniqueLabelMap(getClass());
+
    private ModelInstance pickPointSphere;
 
-   private ArrayList<ModelInstance> leftSpheres = new ArrayList<>();
-   private ArrayList<ModelInstance> rightSpheres = new ArrayList<>();
+   private final ArrayList<ModelInstance> leftSpheres = new ArrayList<>();
+   private final ArrayList<ModelInstance> rightSpheres = new ArrayList<>();
 
-   private SideDependentList<FramePose3D> stancePoses = new SideDependentList<>();
-   private final FramePose3D latestPose = new FramePose3D(ReferenceFrame.getWorldFrame());
-   private final SideDependentList<RDXFootstepGraphic> footstepGraphics;
+   public SideDependentList<FramePose3D> stancePoses = new SideDependentList<>();
+   private final FramePose3D latestPickPoint = new FramePose3D(ReferenceFrame.getWorldFrame());
```

Comment:
does this name convention change imply that it will always be selected? How does the pose get set programmatically (like from a message)?

Reply:
Its only selected when the RDXStanceSelectionPanel is active, the latestPickPoint pose gets set programmatically from the pickPointSphere, and when the left click on the mouse happens, its sent as a message to the ContinuousHiking process